diff -r 682082a4d5b5 mozilla/calendar/base/src/calRecurrenceInfo.js
--- a/mozilla/calendar/base/src/calRecurrenceInfo.js	Sun Sep 21 13:02:58 2008 -0400
+++ b/mozilla/calendar/base/src/calRecurrenceInfo.js	Sun Sep 21 13:08:11 2008 -0400
@@ -487,6 +487,13 @@ calRecurrenceInfo.prototype = {
         }
 
         var startDate = this.mBaseItem.recurrenceStartDate;
+        if (startDate == null) {
+            // Todo created by other apps may have a saved recurrence but
+            // start and due dates disabled.  Since no recurrenceStartDate,
+            // treat as undated task.
+            return [];
+        }
+
         var dates = [];
 
         // toss in exceptions first. Save a map of all exceptions ids, so we
