diff --git a/calendar/base/content/calendar-task-view.js b/calendar/base/content/calendar-task-view.js
--- a/calendar/base/content/calendar-task-view.js
+++ b/calendar/base/content/calendar-task-view.js
@@ -139,9 +139,10 @@
                 parentItem = parentItem.parentItem;
             }
             var recurrenceInfo = parentItem.recurrenceInfo;
-            if (displayElement("calendar-task-details-repeat-row", recurrenceInfo != null)) {
+            var recurStart = parentItem.recurrenceStartDate;
+            if (displayElement("calendar-task-details-repeat-row", recurrenceInfo && recurStart)) {
                 var kDefaultTimezone = calendarDefaultTimezone();
-                var startDate = item.entryDate ? item.entryDate.getInTimezone(kDefaultTimezone) : null;
+                var startDate = recurStart.getInTimezone(kDefaultTimezone);
                 var endDate = item.dueDate ? item.dueDate.getInTimezone(kDefaultTimezone) : null;
                 var detailsString = recurrenceRule2String(recurrenceInfo,startDate,endDate,startDate.isDate);
                 if (detailsString) {
