diff --git a/calendar/base/src/calItemBase.js b/calendar/base/src/calItemBase.js
--- a/calendar/base/src/calItemBase.js
+++ b/calendar/base/src/calItemBase.js
@@ -415,7 +415,7 @@
         } else {
             this.modify();
         }
-        if (aValue || !isNaN(aValue)) {
+        if (aValue || !isNaN(parseInt(aValue, 10))) {
             this.mProperties.setProperty(aName.toUpperCase(), aValue);
         } else {
             this.deleteProperty(aName);
