diff --git a/calendar/base/content/dialogs/calendar-event-dialog.js b/calendar/base/content/dialogs/calendar-event-dialog.js
--- a/calendar/base/content/dialogs/calendar-event-dialog.js
+++ b/calendar/base/content/dialogs/calendar-event-dialog.js
@@ -2167,7 +2167,16 @@
     var listener = {
         onOperationComplete: function(aCalendar, aStatus, aOpType, aId, aItem) {
             if (Components.isSuccessCode(aStatus)) {
-                window.calendarItem = aItem;
+                if (window.calendarItem.recurrenceId) {
+                    // We are editing an occurrence. Make sure that the returned
+                    // item is the same occurrence, not its parent item.
+                    let occ = aItem.recurrenceInfo
+                                   .getOccurrenceFor(window.calendarItem.recurrenceId);
+                    window.calendarItem = occ;
+                } else {
+                    // We are editing the parent item, no workarounds needed
+                    window.calendarItem = aItem;
+                }
             }
         }
     };
