diff --git a/calendar/base/content/dialogs/calendar-event-dialog.js b/calendar/base/content/dialogs/calendar-event-dialog.js
--- a/calendar/base/content/dialogs/calendar-event-dialog.js
+++ b/calendar/base/content/dialogs/calendar-event-dialog.js
@@ -1884,7 +1884,7 @@ function saveItem() {
     }
 
     item.removeAllAttendees();
-    if (window.attendees) {
+    if (window.attendees && (window.attendees.length > 0)) {
         for each (var attendee in window.attendees) {
            item.addAttendee(attendee);
         }
